providers/teams/pages/members/[id].tsx

```diff
@@ -0,0 +1,215 @@
+import React, { ChangeEvent, useEffect, useState } from 'react'
+import { useRouter } from "next/router"
+import dynamic from "next/dynamic"
+import { gql } from "apollo-boost"
+import { apolloClient } from '../../lib/apollo'
+
+
+const getTeamMembers = gql`
+query getTeamMembers($teamId: Int!) {
+    team_members(where: {teamId: {_eq: $teamId}}) {
+        teamId
+        user {
+            id
+            email
+            name
+            role
+        }
+    }
+}
+    
+`
+
+const getUserByMail = gql`
+query getUsers($userMail: String!) {
+    users(where: {email: {_eq: $userMail }}) {
+        id
+    }
+}
+`
+
+
+const hasTeam = gql`
+query hasTeam($userId: Int!) {
+    team_members(where: { userId: { _eq: $userId } }) {
+        teamId
+    }
+}
+`
+
+const insertTeamMember = gql`
+mutation InsertTeamMembers($objects: [team_members_insert_input!]!) {
+    insert_team_members(objects: $objects) {
+      affected_rows
+    }
+  }
+`
+
+
+const deleteTeamMember = gql`
+mutation DeleteTeamMembers($uid: Int!, $tid: Int!) {
+    delete_team_members(where: {_and: {userId: {_eq: $uid}, teamId: {_eq: $tid}}}) {
+      returning {
+        user {
+          id
+          email
+        }
+        teamId
+      }
+    }
+  }
+`
+
+async function checkHasTeam(userId: string) {
+    const graphqlMutation = await apolloClient.query({
+        query: hasTeam,
+        variables: { userId }
+    })
+
+    if (graphqlMutation.data.team_members && graphqlMutation.data.team_members.length) {
+        return graphqlMutation.data.team_members[0].teamId
+    }
+
+    return null
+}
+
+
+const addMember = async (teamId: string, userMail: string) => {
+    const { loading, data, errors } = await apolloClient.query({
+        query: getUserByMail,
+        variables: { userMail }
+    })
+    if (loading) return <div>Loading...</div>
+    if (errors || !data) console.error(errors || "Couldn't retrieve data.")
+
+    if (data.users.length > 0) {
+        const userId = data.users[0].id
+        console.log(`userId ${userId}`);
+
+
+        const hasTeamId = await checkHasTeam(userId)
+        if (hasTeamId) {
+            console.log(`${userMail} is already in team ${hasTeamId}`);
+        } else {
+            // add user to team
+            const { data, errors } = await apolloClient.mutate({
+                mutation: insertTeamMember,
+                variables: { objects: { teamId: parseInt(teamId), userId } }
+            })
+            if (errors) {
+                alert(errors[0])
+                throw errors[0]
+            }
+            return 'success'
```

Comment:
Just a note, if you want to use a non false value returned to check whether addMember succeded or not, I'd expect to return the object of the new user, but that's up to you :)